### PR TITLE
[NEW] Delete Teams

### PR DESCRIPTION
**Pull request by gerzonc**
## Proposed changes

Adds the functionality to remove a team along with its channels.

Depends on #3117 

## Screenshots

<table>
  <thead>
    <tr>
      <th>Without channels</th>
      <th>With channels</th>
      </thead>
    </tr>
    <tbody>
      <tr>
        <td>
        image
        </td>
        <td>
        image
        </td>
      </tr>
    </tbody>
</table >

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves a current function)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/views/RoomInfoEditView/index.js [hunk 1]
```diff
@@ -284,6 +292,72 @@ class RoomInfoEditView extends React.Component {
...
+							title: I18n.t('Confirmation'),
+							message: I18n.t('You_are_deleting_the_team', { team: RocketChat.getRoomTitle(room) }),
+							confirmationText: I18n.t('Yes_action_it', { action: I18n.t('delete') }),
+							onPress: data => this.handleDeleteTeam(data)
+						});
+					}
+				});
+			} else {
+				showConfirmationAlert({
+					title: I18n.t('Confirmation'),
```
Comment (diegolmello) [inline app/views/RoomInfoEditView/index.js:346]: Can we use the default from `showConfirmationAlert` instead?

File: app/views/RoomInfoEditView/index.js [hunk 2]
```diff
@@ -284,6 +292,72 @@ class RoomInfoEditView extends React.Component {
...
+					message: I18n.t('You_are_deleting_the_team', { team: RocketChat.getRoomTitle(room) }),
+					confirmationText: I18n.t('Yes_action_it', { action: I18n.t('delete') }),
+					onPress: () => this.handleDeleteTeam()
+				});
+			}
+		} catch (e) {
+			showConfirmationAlert({
```
Comment (diegolmello) [inline app/views/RoomInfoEditView/index.js:353]: Can you explain why we're calling this here?
If `this.handleDeleteTeam` throws, is this going to run again?

File: app/views/RoomInfoEditView/index.js [hunk 3]
```diff
@@ -284,6 +292,72 @@ class RoomInfoEditView extends React.Component {
...
+				navigation.navigate('SelectListView', {
+					title: 'Delete_Team',
+					data: teamChannels,
+					infoText: 'Select_channels_to_delete',
+					nextAction: () => {
+						showConfirmationAlert({
+							title: I18n.t('Confirmation'),
+							message: I18n.t('You_are_deleting_the_team', { team: RocketChat.getRoomTitle(room) }),
+							confirmationText: I18n.t('Yes_action_it', { action: I18n.t('delete') }),
+							onPress: data => this.handleDeleteTeam(data)
```
Comment (diegolmello) [inline app/views/RoomInfoEditView/index.js:340]: This is never sending `data` to `handleDeleteTeam`.

**Review by diegolmello — CHANGES_REQUESTED** [review]

File: app/views/RoomInfoEditView/index.js (hunk 2 above)
Comment (gerzonc) [reply app/views/RoomInfoEditView/index.js:353]: This is wrong, fixed

**Review by diegolmello — APPROVED** [review]

**Pull request merged**